tools/istio-iptables/pkg/dependencies/implementation_linux.go

```diff
@@ -112,6 +114,12 @@ func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, args ..
 		stderr = &bytes.Buffer{}
 		externalCommand.Stdout = stdout
 		externalCommand.Stderr = stderr
+		externalCommand.Stdin = stdin
+		if stdin != nil {
+			if _, err := stdin.Seek(0, io.SeekStart); err != nil {
```

Comment:
Am I reading this back that we may use the stdin results multiple times, so we are seeking back to the start of stdin?

If so, TIL - I didn't know you could do this

Reply:
Yeah, this code is wrapped in a `RetryWithContext`, so it must reset the `Reader` on every attempt. AFAICT, the way to do this is by using `ReadSeeker` and calling `Seek()`.

Reply:
This may be a naive question but can you really seek back on stdin in linux in general? How does that work with things like `yes | cat` or whatever where the input is ~infinite?

Reply:
No, I don't think you can do that in general. But here we're only rewinding before we pass the reader to the process as stdin.

Reply:
Ohhh, makes perfect sense - thanks